### PR TITLE
feat: cnn support for recurrent systems

### PR DESCRIPTION
**Pull request by SimonDuToit**
Previously it was not possible to use a CNN in Mava's recurrent systems. This PR makes CNNs compatible with recurrent systems and adds a relevant config for a recurrent CNN network.

### CODE REVIEW

File: mava/networks.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __call__(self, observation: chex.Array) -> chex.Array:
             x = self.activation_fn(x)
 
         # Reshape should keep the batch and agent dimensions unchanged.
-        return x.reshape((x.shape[0], x.shape[1], -1))
+        return jax.lax.collapse(x, -3)
```
Comment (sash-a) [inline mava/networks.py:85]: I'd say remove the comment above for these two
```suggestion
        # Collapse (merge) the last three dimensions (width, height, channels)
        # Leave the batch, agent and time (if recurrent) dims unchanged.
        return jax.lax.collapse(x, -3)
```

**Review by sash-a — CHANGES_REQUESTED** [review]
Small change otherwise looks good. Can we just wait till we've run a couple experiments on connector to see that the recurrent systems perform ok :pray:

**Review by sash-a — APPROVED** [review]
Thanks @SimonDuToit :fire:

**Review by OmaymaMahjoub — APPROVED** [review]
Thanks @SimonDuToit :fire:

**Pull request merged**